trie/codec.go

```diff
@@ -20,15 +20,58 @@ func KeyEncode(k []byte) []byte {
 
 // keyToHex turns bytes into nibbles
 func keyToHex(in []byte) []byte {
-	l := len(in)*2 + 1
+	l := len(in) * 2
 	res := make([]byte, l)
 	for i, b := range in {
 		res[2*i] = b / 16
 		res[2*i+1] = b % 16
 	}
 
-	// last index of nibble array is set to 16 due to the way branches are indexed
-	// branch at 0...15 points to possible children, branch at 16 is the value at the branch
-	res[l-1] = 16
 	return res
 }
+
+// hexToKey performs the opposite of keyToHex; turns nibbles back into bytes
+// removes last byte if length of input is odd (set to 16 if using keyToHex)
+func hexToKey(in []byte) []byte {
+	l := len(in) / 2
+	res := make([]byte, l)
+	for i := 0; i < len(in)-1; i = i + 2 {
+		res[i/2] = in[i+1] | in[i]<<4
+	}
+	return res
+}
+
+// bigKeySize returns the node type's BigKeySize
+// BigKeySize is 125 if node is extension, 126 if node is leaf
+func bigKeySize(n node) int {
```

Comment:
I don't see this being used anywhere

Reply:
it will be used in the encoding of the nodes. I implemented this already when I was working on encoding for the previous tree so I figured we could leave it

Reply:
commented it out for now to appease the ci